File: Part 2 - Regression/Simple Linear Regression/simple_linear_regression.py
# -*- coding: utf-8 -*-
#Simple Linear Regression


import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing dataset
dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:,:-1].values
Y = dataset.iloc[:,1].values

#spliting the dataset in to training set and dataset
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=1/3,random_state=0)

#Feature scaling 
# from sklearn.preprocessing import StandardScaler
# sc_X = StandardScaler()
# X_train = sc_X.fit_transform(X_train)
# X_test = sc_X.transform(X_test)

#Fitting Simple Linear Regression in to training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, Y_train)

#predict the test result set
Y_pred = regressor.predict(X_test)

#Visualizing the training set results
plt.scatter(X_train, Y_train,color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience(Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show();

#Visualizing the Test set results
plt.scatter(X_test, Y_test,color='red')
plt.plot(X_test, Y_pred, color='blue')
plt.title('Salary vs Experience(Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show();

File: Part 2 - Regression/Simple Linear Regression/simple_linear_regression.R
#Simple Linear Regression

#importing datset
dataset = read.csv('Salary_Data.csv')

#Splitting the dataset in to Traing set and Test set
# install.packages('caTools') 
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset,split == TRUE)
test_set = subset(dataset,split == FALSE)

#Fitting simple linear regression in to traing set
regressor = lm(formula = Salary ~ YearsExperience, 
               data = training_set)

#Predicting the Test set results
y_pred = predict(regressor,newdata = test_set)

#Visualizing the Training set results 
install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x=training_set$YearsExperience, y=training_set$Salary), 
             colour='red')+
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor,newdata = training_set)),
            colour='blue')+
  ggtitle('Salary Vs Experinece (Training set)')+
  xlab('Years of Experience')+
  ylab('Salary')

#Visualizing the Test set results 
install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x=test_set$YearsExperience, y=test_set$Salary), 
             colour='red')+
  geom_line(aes(x=training_set$YearsExperience, y=predict(regressor,newdata = training_set)),
            colour='blue')+
  ggtitle('Salary Vs Experinece (Test set)')+
  xlab('Years of Experience')+
  ylab('Salary')
  